pkg/apis/serving/v1/revision_validation.go

```diff
@@ -179,22 +181,43 @@ func validateTimeoutSeconds(ctx context.Context, timeoutSeconds int64) *apis.Fie
 	return nil
 }
 
-// validateQueueSidecarAnnotation validates QueueSideCarResourcePercentageAnnotation
-func validateQueueSidecarAnnotation(m map[string]string) *apis.FieldError {
+// validateQueueSidecarResourceAnnotations validates QueueSideCarResourcePercentageAnnotation and other QP resource related annotations.
+func validateQueueSidecarResourceAnnotations(m map[string]string) *apis.FieldError {
 	if len(m) == 0 {
 		return nil
 	}
-	k, v, ok := serving.QueueSidecarResourcePercentageAnnotation.Get(m)
-	if !ok {
-		return nil
+
+	var errs *apis.FieldError
+	if k, v, ok := serving.QueueSidecarResourcePercentageAnnotation.Get(m); ok {
+		errs = apis.ErrGeneric("Queue proxy resource percentage annotation is deprecated. Please use the available annotations to explicitly set resource values per service").ViaKey(k).At(apis.WarningLevel)
+		value, err := strconv.ParseFloat(v, 64)
+		if err != nil {
+			errs = errs.Also(apis.ErrInvalidValue(v, apis.CurrentField).ViaKey(k))
+		} else {
+			if value < 0.1 || value > 100 {
+				errs = errs.Also(apis.ErrOutOfBoundsValue(value, 0.1, 100.0, apis.CurrentField).ViaKey(k))
+			}
+		}
 	}
-	errs := apis.ErrGeneric("Queue proxy resource percentage annotation is deprecated. Please use the available annotations to explicitly set resource values per service").ViaKey(k).At(apis.WarningLevel)
-	value, err := strconv.ParseFloat(v, 64)
-	if err != nil {
-		return errs.Also(apis.ErrInvalidValue(v, apis.CurrentField).ViaKey(k))
+	validate := func(k, v string) *apis.FieldError {
+		if _, err := resource.ParseQuantity(v); err != nil {
+			return errs.Also(apis.ErrInvalidValue(v, apis.CurrentField).ViaKey(k))
+		}
+		return nil
 	}
```

Comment:
Can we just move this into the loop below? so we're not creating a closure on every validate invocation?